lib/s3_helper.rb

```diff
@@ -244,15 +244,21 @@ def list(prefix = "", marker = nil)
     s3_bucket.objects(options)
   end
 
-  def tag_file(key, tags)
-    tag_array = []
-    tags.each { |k, v| tag_array << { key: k.to_s, value: v.to_s } }
+  def upsert_tag(key, tag_key:, tag_value:)
+    tags = s3_resource.client.get_object_tagging(bucket: @s3_bucket_name, key:).tag_set
+    tag_index = tags.find_index { |tag| tag[:key].to_s == tag_key.to_s }
```

Comment:
Why are there so many `.to_s` here? Are the keys not strings?

Reply:
They should be strings but we don't have a good way to enforce types so I've added `to_s` out of precaution given that the previous implementation did so as well.